packages/monaco-editor-wrapper/src/editorClassic.ts

```diff
@@ -15,41 +15,54 @@ export type MonacoLanguageExtensionConfig = {
     mimetypes?: string[];
 }
 
-export type EditorClassicConfig = {
+export type EditorAppConfigClassic = {
+    editorAppType: 'classic';
     languageExtensionConfig?: MonacoLanguageExtensionConfig;
     languageDef?: languages.IMonarchLanguage;
     themeData?: editor.IStandaloneThemeData;
 }
 
-export class EditorClassic extends MonacoEditorBase implements MonacoEditorWrapper {
+export class EditorAppClassic extends EditorAppBase implements MonacoEditorWrapper {
+
+    static APP_TYPE = 'classic';
+
+    static createEmptyConfig() {
+        return {
+            editorAppType: EditorAppClassic.APP_TYPE
+        } as EditorAppConfigClassic;
+    }
+
+    getAppType() {
+        return EditorAppClassic.APP_TYPE;
+    }
 
     async init() {
-        const wrapperConfig = this.monacoConfig as EditorClassicConfig;
+        const wrapperConfig = this.editorAppConfig === undefined ? EditorAppClassic.createEmptyConfig() : this.editorAppConfig as EditorAppConfigClassic;
 
         // register own language first
         const extLang = wrapperConfig?.languageExtensionConfig;
         if (extLang) {
             languages.register(extLang);
         }
 
-        const languageRegistered = languages.getLanguages().filter(x => x.id === this.editorConfig.languageId);
+        const languageRegistered = languages.getLanguages().filter(x => x.id === this.editorContentConfig.languageId);
         if (languageRegistered.length === 0) {
             // this is only meaningful for languages supported by monaco out of the box
             languages.register({
-                id: this.editorConfig.languageId
+                id: this.editorContentConfig.languageId
             });
         }
 
         // apply monarch definitions
         const tokenProvider = wrapperConfig?.languageDef;
         if (tokenProvider) {
-            languages.setMonarchTokensProvider(this.editorConfig.languageId, tokenProvider);
+            languages.setMonarchTokensProvider(this.editorContentConfig.languageId, tokenProvider);
         }
         const themeData = wrapperConfig?.themeData;
         if (themeData) {
-            editor.defineTheme(this.editorConfig.theme, themeData);
+            editor.defineTheme(this.editorContentConfig.theme!, themeData);
```

Comment:
Same to assume theme is defined?

Reply:
The constructor ensure it is set, but compiler sees just that it may be undefined as well.